src/main.ts

```diff
@@ -26,17 +26,22 @@ export const run = async () => {
         let inlineScript: string = core.getInput('inlineScript', { required: true });
         let azcliversion: string = core.getInput('azcliversion', { required: false }).trim().toLowerCase();
 
-        if(azcliversion == AZ_CLI_VERSION_DEFAULT_VALUE){
-            const { stdout, stderr } = await cpExec('az version');
-            if (!stderr) {
-                try {
-                    azcliversion = JSON.parse(stdout)["azure-cli"]
-                }
-                catch (er) {
+        if (azcliversion == AZ_CLI_VERSION_DEFAULT_VALUE) {
+            try {
+                const { stdout, stderr } = await cpExec('az version');
+                if (!stderr) {
+                    try {
+                        azcliversion = JSON.parse(stdout)["azure-cli"]
+                    }
+                    catch (er) {

```

Comment:
You don't need inner try catch now as error will be caught by upper try...catch. This will reduce code duplicacy.

Reply:
Done